modules/mysql/src/main/java/org/testcontainers/containers/MySQLContainerProvider.java

```diff
@@ -1,9 +1,26 @@
 package org.testcontainers.containers;
 
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.testcontainers.jdbc.ConnectionUrl;
+
 /**
  * Factory for MySQL containers.
  */
 public class MySQLContainerProvider extends JdbcDatabaseContainerProvider {
+  
+    /**
+     * Groups URL of format "jdbc:tc:(databaseType):(optinal_image_tag)//(hostanme)(optional :(numeric_port))/(databasename)(?parameters)"
```

Comment:
A couple of typos in here (`optinal` and `hostanme`)

Reply:
Typos are corrected now.